airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java

```diff
@@ -84,11 +132,39 @@ public List<String> getValidationMessagePaths(final JsonNode schemaJson, final J
         .collect(Collectors.toList());
   }
 
+  public void ensure(final JsonNode schemaJson, final JsonNode objectJson) throws JsonValidationException {
+    final Set<ValidationMessage> validationMessages = validateInternal(schemaJson, objectJson);
+    if (validationMessages.isEmpty()) {
+      return;
+    }
+
+    throw new JsonValidationException(String.format(
+        "json schema validation failed when comparing the data to the json schema. \nErrors: %s \nSchema: \n%s",
+        Strings.join(validationMessages, ", "),
+        schemaJson.toPrettyString()));
+  }
+
+  public void ensureAsRuntime(final JsonNode schemaJson, final JsonNode objectJson) {
```

Comment:
I know this is an existing method, and I'm not arguing for changing it here, but this sort of pattern is a smell to me.

Reply:
agree. all of this should be client side code since it makes the class methods shallow.

in a follow up PR I plan to refactor all of this including the existing `validate`, `getValidationMessageArgs` and `getValidationMessagePaths` methods - they all suffer from being shallow methods.